File: app/src/main/java/ee/it/trailers/MainActivity.java
package ee.it.trailers;

import android.app.Fragment;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import ee.it.trailers.tmdb.Movie;

public class MainActivity extends AppCompatActivity implements MoviesFragment.OnMovieSelected {
    private Toolbar mToolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        super.setContentView(R.layout.activity_main);

        mToolbar = (Toolbar) findViewById(R.id.my_toolbar);
        setSupportActionBar(mToolbar);

        if (savedInstanceState == null) {
            //TrailersFragment fragment = new TrailersFragment();
            MoviesFragment fragment = new MoviesFragment();
            getFragmentManager().beginTransaction()
                    .add(R.id.fragment_container, fragment)
                    .commit();
        }
    }

    @Override
    public void onBackPressed() {
        final android.app.FragmentManager fm = getFragmentManager();
        int count = fm.getBackStackEntryCount();
        if (count > 0) {
            if (count == 1) {
                mToolbar.setTitle(R.string.app_name);
            }

            fm.popBackStack();
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public void onMovieSelected(Movie movie) {
        mToolbar.setTitle(movie.title);

        final Fragment fragment = MovieDetailsFragment.newInstance(movie.id);
        getFragmentManager().beginTransaction()
                .replace(R.id.fragment_container, fragment)
                .addToBackStack(null)
                .commit();
    }
}

File: app/src/main/java/ee/it/trailers/MainActivity.kt
package ee.it.trailers

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (intent?.action?.equals(Intent.ACTION_VIEW)?:false) {
            val uri = intent.data
            sendToPulsar(uri.toString())
            finish()
            return
        }

        super.setContentView(R.layout.activity_main)
        setSupportActionBar(my_toolbar)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                    .add(R.id.fragment_container, MoviesFragment())
                    .commit()
        }
    }

    override fun onBackPressed() {
        val fm = fragmentManager
        if (fm.backStackEntryCount > 0) {
            fm.popBackStack()
        } else {
            super.onBackPressed()
        }
    }

    fun sendToPulsar(magnet: String) {
        var app = application as MyApplication
        var pulsar = Pulsar(this, app.httpClient)
        pulsar.playUri(magnet)
        Toast.makeText(this, R.string.magnet_sent, Toast.LENGTH_SHORT)
                .show()
    }
}
